x-pack/test/apm_api_integration/tests/alerts/transaction_duration.spec.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AggregationType, ApmRuleType } from '@kbn/apm-plugin/common/rules/apm_rule_types';
+import { apm, timerange } from '@kbn/apm-synthtrace-client';
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import {
+  createApmRule,
+  createIndexConnector,
+  fetchServiceInventoryAlertCounts,
+  fetchServiceTabAlertCount,
+} from './alerting_api_helper';
+import {
+  waitForRuleStatus,
+  waitForDocumentInIndex,
+  waitForAlertInIndex,
+} from './wait_for_rule_status';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const apmApiClient = getService('apmApiClient');
+  const esDeleteAllIndices = getService('esDeleteAllIndices');
+
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  registry.when('transaction duration alert', { config: 'basic', archives: [] }, () => {
+    let ruleId: string;
+    let actionId: string | undefined;
+
+    const INDEX_NAME = 'transaction-duration';
+
+    before(async () => {
+      const opbeansJava = apm
+        .service({ name: 'opbeans-java', environment: 'production', agentName: 'java' })
+        .instance('instance');
+      const opbeansNode = apm
+        .service({ name: 'opbeans-node', environment: 'production', agentName: 'node' })
+        .instance('instance');
+      const events = timerange('now-15m', 'now')
+        .ratePerMinute(1)
+        .generator((timestamp) => {
+          return [
+            opbeansJava
+              .transaction({ transactionName: 'tx-java' })
+              .timestamp(timestamp)
+              .duration(5000)
+              .success(),
+            opbeansNode
+              .transaction({ transactionName: 'tx-node' })
+              .timestamp(timestamp)
+              .duration(4000)
+              .success(),
+          ];
+        });
+      await synthtraceEsClient.index(events);
+    });
+
+    after(async () => {
+      await synthtraceEsClient.clean();
+      await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'foo');
+      await supertest.delete(`/api/actions/connector/${actionId}`).set('kbn-xsrf', 'foo');
+      await esDeleteAllIndices([INDEX_NAME]);
+      await es.deleteByQuery({
+        index: '.alerts*',
```

Comment:
Perhaps extract it and make it more specific:

```ts
const APM_ALERTS_INDEX = '.alerts-observability.apm.alerts-default';
```

```suggestion
        index: APM_ALERTS_INDEX,
```

Reply:
done